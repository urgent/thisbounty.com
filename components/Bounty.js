import React from "react";
import { Grid, Image, Text, Box } from "@chakra-ui/react";
import Life from "./Life";
import Link from "next/link";
import Back from "./Back";

function back({ parent, id }) {
  // parent id specified `/bounty/${parent.wd}`
  if (parent?.wd) {
    return <Back text="Back" href={`/bounty/${parent.wd}`} id={id} />;
  }
  // null parent id, redirect to home
  else if (parent?.wd === null) {
    return <Back text="Back" href={`/`} id={id} />;
  }
  // no parent specified, don't show back button.
  return <></>;
}

function name({ title, id, child }) {
  if (child?.length) {
    return (
      <Link href={`/bounty/${id}`}>
        <Text
          data-cy={`bounty-${id}-title`}
          as="h3"
          sx={{ gridRow: "title" }}
          fontSize="xs"
          cursor="pointer"
          textDecoration="underline"
          fontWeight="bold"
        >
          {title}
        </Text>
      </Link>
    );
  }
  return (
    <Text
      data-cy={`bounty-${id}-title`}
      as="h3"
      sx={{ gridRow: "title" }}
      fontSize="xs"
    >
      {title}
    </Text>
  );
}

export default function Bounty({
  id,
  img,
  title,
  damage,
  level,
  child,
  parent,
}) {
  return (
    <Grid
      data-cy={`bounty-${id}`}
      templateColumns="[back] 20px [image] 2fr [stats] 3fr"
      w={300}
      h={140}
      my={1}
      key={id}
    >
      {back({ parent, id })}
      <Image
        data-cy={`bounty-${id}-image`}
        src={img}
        sx={{ gridColumn: "image" }}
      />
      <Grid gridColumn="stats" templateRows="[title] 1em [life] 1em" ml={2}>
        {name({ title, id, child })}
        <Box sx={{ gridRow: "life" }}>
          <Life level={level} damage={damage} />
        </Box>
      </Grid>
    </Grid>
  );
}
