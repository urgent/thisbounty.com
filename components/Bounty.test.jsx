import React from "react";
import renderer from "react-test-renderer";
import Bounty from "./Bounty";

it("renders correctly", () => {
  const tree = renderer.create(<Bounty />).toJSON();
  expect(tree).toMatchSnapshot();
});
