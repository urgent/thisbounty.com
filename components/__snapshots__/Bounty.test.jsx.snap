// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-1mr5rgs"
  data-cy="bounty-undefined"
>
  <img
    className="chakra-image__placeholder css-jcgxpa"
  />
  <div
    className="css-1l0lp1p"
  >
    <h3
      className="chakra-text css-jjbsct"
    />
    <p
      className="chakra-text css-vr7x6g"
    >
      Life
    </p>
  </div>
</div>
`;
